src/vocgui/views/training_set_by_id_viewset.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework import viewsets
+from django.db.models import Count, Q
+from vocgui.models import TrainingSet
+from vocgui.serializers import TrainingSetSerializer
+from vocgui.models import Discipline
+from .utils import check_group_object_permissions
+
+
+class TrainingSetByIdViewSet(viewsets.ModelViewSet):
+    """
+    Defines a view set for a TrainingSet module of a given id.
+    Inherits from `viewsets.ModelViewSet` and defines queryset
+    and serializers.
+    """
+
+    serializer_class = TrainingSetSerializer
+    http_method_names = ["get"]
+
+    def get_queryset(self):
+        """
+        Defining custom queryset
+
+        :param self: A handle to the :class:`TrainingSetViewSet`
+        :type self: class
+
+        :return: (filtered) queryset
+        :rtype: QuerySet
+        """
+        if getattr(self, "swagger_fake_view", False):
+            return TrainingSet.objects.none()
+        user = self.request.user
+        queryset = TrainingSet.objects.filter(
+            id=self.kwargs["training_set_id"], released=True
+        ).annotate(
```

Comment:
```suggestion
        queryset = TrainingSet.objects.filter(released=True).annotate(
```
We don't need to filter for the id, the model view set does this automatically for us.

Reply:
Yes, the rest framework automatically generates two routes - one for the list view and one for the detail view (see [docs](https://www.django-rest-framework.org/api-guide/routers/#usage)).
When you received all training sets, you probably still had the regex pattern `/(?P<training_set_id>[0-9]+)` in urls.py - which means that the id parameter is still automatically created, but it's the second parameter after the endpoint, and the first parameter is just ignored...
Filtering the queryset explicitly by the `training_set_id` parameter works as well, it just has the weird side-effect that the detail view returns a list with exactly one element and when the id is appended twice, the requested training set is returned as a single object:

`GET /api/training_set_details/5/`

```
[
    {
        "id": 5,
        "title": "Grundlagen Streichen",
        "description": "Grundlegende Vokabeln zu Streichen",
        "icon": "http://127.0.0.1:8000/media/images/001-roller.png",
        "total_documents": 12
    }
]
```


`GET /api/training_set_details/5/5/`

```
{
    "id": 5,
    "title": "Grundlagen Streichen",
    "description": "Grundlegende Vokabeln zu Streichen",
    "icon": "http://127.0.0.1:8000/media/images/001-roller.png",
    "total_documents": 12
}
```

Reply:
Thanks for explanation. I removed it, but I could not get it working without checking for the id, because I wanted to check the permissions and therefore needed to manually get and check for the group id. Only other solution that came to my mind, would be to have to database request one with and one without the check, but this would be even worse?